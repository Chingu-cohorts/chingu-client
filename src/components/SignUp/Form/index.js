import React, { Component } from "react";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import { Form, Icon, Input, Button } from "antd";

import "./form.css";

const FormItem = Form.Item;

class SignUpForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      first_name: "",
      last_name: "",
      email: "",
      password: ""
    };
  }

  handleSignUp = e => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      if (!err) {
        let { email, first_name, last_name, github_url, password } = values;
        console.log(values);

        this.props
          .mutate({
            variables: {
              email,
              first_name,
              last_name,
              github_url,
              password
            }
          })
          .then(({ data }) => {
            console.log(data);
            window.location = "/";
          })
          .catch(err => {
            console.error(err);
          });
      }
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <Form onSubmit={this.handleSignUp} className="signup-form">
        <FormItem>
          {getFieldDecorator("email", {
            rules: [{ required: true, message: "Please input your email!" }]
          })(
            <Input
              prefix={<Icon type="global" style={{ fontSize: 13 }} />}
              type="email"
              placeholder="Email"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("first_name", {
            rules: [
              { required: true, message: "Please input your first name!" }
            ]
          })(
            <Input
              prefix={<Icon type="user" style={{ fontSize: 13 }} />}
              type="text"
              placeholder="First name"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("last_name", {
            rules: [{ required: true, message: "Please input your last name!" }]
          })(
            <Input
              prefix={<Icon type="team" style={{ fontSize: 13 }} />}
              type="text"
              placeholder="Last name"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("github_url", {
            rules: [
              { required: true, message: "Please input your GitHub URL!" }
            ]
          })(
            <Input
              prefix={<Icon type="github" style={{ fontSize: 13 }} />}
              type="text"
              placeholder="GitHub URL"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("password", {
            rules: [{ required: true, message: "Please input your password!" }]
          })(
            <Input
              prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
              type="password"
              placeholder="Password"
            />
          )}
        </FormItem>
        <FormItem>
          <Button
            type="primary"
            htmlType="submit"
            className="signup-form-button"
          >
            Apply
          </Button>
        </FormItem>
      </Form>
    );
  }
}

const registerUser = gql`
  mutation registerUser(
    $first_name: String!
    $last_name: String!
    $email: String!
    $github_url: String!
    $password: String!
  ) {
    createUser(
      user_data: {
        first_name: $first_name
        last_name: $last_name
        github_url: $github_url
      }
      password: $password
      email: $email
    ) {
      email
    }
  }
`;

const wrappedForm = Form.create()(SignUpForm);

export default graphql(registerUser)(wrappedForm);
