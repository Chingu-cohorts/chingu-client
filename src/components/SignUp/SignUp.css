.signup {
  background: url(/img/signup.png) repeat center center fixed;
  padding: 1rem;
}

.signup .signup-form {
  box-shadow: 2px 2px 6px 0px #bdbdbd;
  margin: 3rem 0;
  padding: 0 2rem;
}
