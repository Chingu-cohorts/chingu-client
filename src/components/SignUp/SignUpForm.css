.signup-form {
  max-width: 400px;
}

.signup-form-button {
  width: 100%;
}
