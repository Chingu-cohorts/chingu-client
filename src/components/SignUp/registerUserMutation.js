import gql from "graphql-tag";

const registerUser = gql`
  mutation registerUser(
    $first_name: String!
    $last_name: String!
    $email: String!
    $github_url: String!
    $password: String!
  ) {
    createUser(
      user_data: {
        first_name: $first_name
        last_name: $last_name
        github_url: $github_url
      }
      email: $email
      password: $password
    ) {
      jwt
      user {
        id
      }
    }
  }
`;

export default registerUser;
