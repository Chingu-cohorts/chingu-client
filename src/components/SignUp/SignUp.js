import React, { Component } from "react";

import Form from "./Form";
import "./SignUp.css";

class SignUp extends Component {
  render() {
    return (
      <div className="signup">
        <div className="container">
          <div className="row">
            <div className="col m6 offset-m3 grey lighten-5 signup-form">
              <h1 className="center">Sign Up</h1>
              <Form />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default SignUp;
