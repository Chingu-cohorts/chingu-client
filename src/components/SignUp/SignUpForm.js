import React, { Component } from "react";
import { graphql } from "react-apollo";
import { Form, Icon, Input, Button } from "antd";

import registerUser from "./registerUserMutation";
import "./SignUpForm.css";

const FormItem = Form.Item;

class SignUpForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      first_name: "",
      last_name: "",
      email: "",
      password: "",
      userName: ""
    };
  }

  handleSignUp = e => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      if (!err) {
        let { email, first_name, last_name, github_url, password, userName } = values;

        this.props
          .mutate({
            variables: {
              email,
              first_name,
              last_name,
              github_url,
              password,
              userName,
            }
          })
          .then(({ data }) => {
            window.localStorage.setItem("token", data.createUser.jwt);
            window.location = "/profile/" + data.createUser.user.id;
          })
          .catch(err => {
            console.error(err);
          });
      }
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <Form onSubmit={this.handleSignUp} className="signup-form">
        <FormItem>
          {getFieldDecorator("email", {
            rules: [{ required: true, message: "Please input your email!" }]
          })(
            <Input
              prefix={<Icon type="global" style={{ fontSize: 13 }} />}
              type="email"
              placeholder="Email"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("username", {
            rules: [
              { required: true, message: "Please input your username!" }
            ]
          })(
            <Input
              prefix={<Icon type="user" style={{ fontSize: 13 }} />}
              type="text"
              placeholder="User Name"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("first_name", {
            rules: [
              { required: true, message: "Please input your first name!" }
            ]
          })(
            <Input
              prefix={<Icon type="user" style={{ fontSize: 13 }} />}
              type="text"
              placeholder="First name"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("last_name", {
            rules: [{ required: true, message: "Please input your last name!" }]
          })(
            <Input
              prefix={<Icon type="team" style={{ fontSize: 13 }} />}
              type="text"
              placeholder="Last name"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("github_url", {
            rules: [
              { required: true, message: "Please input your GitHub URL!" }
            ]
          })(
            <Input
              prefix={<Icon type="github" style={{ fontSize: 13 }} />}
              type="text"
              placeholder="GitHub URL"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("password", {
            rules: [{ required: true, message: "Please input your password!" }]
          })(
            <Input
              prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
              type="password"
              placeholder="Password"
            />
          )}
        </FormItem>
        <FormItem>
          {getFieldDecorator("passwordVerify", {
            rules: [{ required: true, message: "Please verify your password!" }]
          })(
            <Input
              prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
              type="password"
              placeholder="Verify Password"
            />
          )}
        </FormItem>
        <FormItem>
          <Button
            type="primary"
            htmlType="submit"
            className="signup-form-button"
          >
            Register
          </Button>
        </FormItem>
      </Form>
    );
  }
}

export const wrappedForm = Form.create()(SignUpForm);

export default graphql(registerUser)(wrappedForm);