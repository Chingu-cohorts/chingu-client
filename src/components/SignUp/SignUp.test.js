import React from "react";
import { shallow } from "enzyme";

import SignUp from "./SignUp";

describe("SignUp Component", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper).toMatchSnapshot();
  });
});
