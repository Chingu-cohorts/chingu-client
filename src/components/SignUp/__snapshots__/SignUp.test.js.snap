// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          style={
                    Object {
                              "textAlign": "center",
                            }
          }
>
          Sign Up
</h1>,
        <Row
          gutter={0}
          justify="center"
          type="flex"
>
          <Col
                    lg={8}
                    md={12}
                    xs={24}
          >
                    <Apollo(Form(WrappedComponent)) />
          </Col>
</Row>,
      ],
      "className": "signup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Col
            lg={8}
            md={12}
            xs={24}
>
            <Apollo(Form(WrappedComponent)) />
</Col>,
          "gutter": 0,
          "justify": "center",
          "type": "flex",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Apollo(Form(WrappedComponent)) />,
            "lg": 8,
            "md": 12,
            "xs": 24,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
                        Object {
                                    "textAlign": "center",
                                  }
            }
>
            Sign Up
</h1>,
          <Row
            gutter={0}
            justify="center"
            type="flex"
>
            <Col
                        lg={8}
                        md={12}
                        xs={24}
            >
                        <Apollo(Form(WrappedComponent)) />
            </Col>
</Row>,
        ],
        "className": "signup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Col
              lg={8}
              md={12}
              xs={24}
>
              <Apollo(Form(WrappedComponent)) />
</Col>,
            "gutter": 0,
            "justify": "center",
            "type": "flex",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Apollo(Form(WrappedComponent)) />,
              "lg": 8,
              "md": 12,
              "xs": 24,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
